pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,214 @@
+    steps: tensor, optional
+        Length of the markov chain. Only needed if state is not provided.
+    init_dist : unnamed distribution, optional
+        Vector distribution for initial values. Unnamed refers to distributions
+        created with the ``.dist()`` API. Distribution should have shape n_states.
+        If not, it will be automatically resized. Defaults to pm.Categorical.dist(p=np.full(n_states, 1/n_states)).
+        .. warning:: init_dist will be cloned, rendering it independent of the one passed as input.
+    """
+
+    rv_type = DiscreteMarkovChainRV
+
+    def __new__(cls, *args, steps=None, **kwargs):
+        # Subtract 1 step to account for x0 given, better match user expectation of
+        # len(markov_chain) = steps
+        if steps is not None:
+            steps -= 1
+
+        steps = get_support_shape_1d(
+            support_shape=steps,
+            shape=None,
+            dims=kwargs.get("dims", None),
+            observed=kwargs.get("observed", None),
+            support_shape_offset=1,
+        )
+
+        return super().__new__(cls, *args, steps=steps, **kwargs)
+
+    @classmethod
+    def dist(cls, P=None, logit_P=None, steps=None, init_dist=None, **kwargs):
+
+        shape = kwargs.get("shape", None)
+
+        steps = get_support_shape_1d(support_shape=steps, shape=shape, support_shape_offset=1)
+
+        batch_size = None
+        if shape is not None:
+            batch_size = shape[1:] if shape[0] == steps else shape
+
+        # TODO: Was getting errors with int32 vs int64 mismatches, is there a better way to address this?
+        dtype = kwargs.get("dtype", None) or pytensor.config.floatX.replace("float", "int")
+
+        if steps is None:
+            raise ValueError("Must specify steps or shape parameter")
+        if P is None and logit_P is None:
+            raise ValueError("Must specify P or logit_P parameter")
+        if P is not None and logit_P is not None:
+            raise ValueError("Must specify only one of either P or logit_P parameter")
+
+        if logit_P is not None:
+            P = pm.math.softmax(logit_P, axis=1)
+
+        P = at.as_tensor_variable(P)
+        steps = at.as_tensor_variable(steps.astype(dtype), ndim=1)
+
+        if init_dist is not None:
+            if not isinstance(init_dist, TensorVariable) or not isinstance(
+                init_dist.owner.op, (RandomVariable, SymbolicRandomVariable)
+            ):
+                raise ValueError(
+                    f"Init dist must be a distribution created via the `.dist()` API, "
+                    f"got {type(init_dist)}"
+                )
+            check_dist_not_registered(init_dist)
+            if init_dist.owner.op.ndim_supp > 1:
+                raise ValueError(
+                    "Init distribution must have a scalar or vector support dimension, ",
+                    f"got ndim_supp={init_dist.owner.op.ndim_supp}.",
+                )
+        else:
+            warnings.warn(
```

Comment:
This initial distribution seems reasonable enough that we might not need the warning?

Reply:
I defer to your judgement. I tend to err on the side of "give too much information" in my code -- reasonable or not, it's something we're injecting into the user's model.

Reply:
No strong opinion, go with your preference

Reply:
Verbose for now then, to match AR. Open to changing it in the future.